### PR TITLE
feat:added layout for feed page and create/delete post functionality

### PR DESCRIPTION
**Pull request by Hardiegogo**
- Added feed page
- Added components sidebar(static), suggestion box(static), postsList
- created thunk functions for create/delete  post functionality

Screenshots:
image

for smaller screens:
image

### CODE REVIEW

File: src/components/Post.jsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import React,{useState} from 'react'
+import {AiOutlineHeart,AiFillDelete,AiFillHeart} from 'react-icons/ai'
+import {BsBookmark,BsThreeDotsVertical} from 'react-icons/bs'
+import {GoComment} from 'react-icons/go'
+import { useSelector,useDispatch } from 'react-redux'
+import { deletePost, dislikePost, likePost } from '../utils/post-utils/post-services'
+
+const isLikedCalculator=(likedBy,username)=>{
+    const result=likedBy.find(user=>user.username===username)
+    return result ? true : false
```
Comment (AmishMishra11) [inline src/components/Post.jsx:10]: You can just return result, no need to check for true or false

File: src/components/PostSection.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        setContent('')
+        dispatch(addPost(post))
+    }
+    return (
+      <div>
+        <div className="flex gap-2">
+          <img
+            className="w-12 rounded-full object-cover h-12 "
+            src={user.userProfile}
+            alt=""
```
Comment (AmishMishra11) [inline src/components/PostSection.jsx:21]: add alt to image

File: src/components/Sidebar.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+          <span className='lg:hidden'>Create a new post</span> +
+        </button>
+            </li>
+        </ul>
+        
+    </div>
+  )
+}
+
+//bg-[#ffbd8b]
```
Comment (AmishMishra11) [inline src/components/Sidebar.jsx:22]: remove unwanted comments

File: src/components/Suggestionbar.jsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import { useSelector } from 'react-redux'
+
+const ProfileChip=({user:{userProfile,username}})=>{
+    
+    return <div className='flex justify-between flex-wrap gap-1'>
+        <div className='flex items-center font-sans gap-1'>
+            <img className='w-12 rounded-full object-cover h-12 ' src={userProfile} alt="" />
```
Comment (AmishMishra11) [inline src/components/Suggestionbar.jsx:8]: Provide alt to image

**Review by AmishMishra11 — APPROVED** [review]
Great UI
Good folder structure
Clean Code

File: src/components/Post.jsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+const Post=({post:{username,content,likes,comments,userProfile,_id}})=> {
+    const user = useSelector(state=>state.auth.loggedInUser)
```
Comment (dev-enforced) [inline src/components/Post.jsx:14]: Can destructure it as making it easy and less to write on the RHS of =
```
const {loggedInUser:user}=useSelector(state=>state.auth)
```

**Review by dev-enforced — APPROVED** [review]
The code structure looks really good !! And good work on creating the async Thunks before hand can help in implementing the features later on as we add in Mircle Project !!
The live preview is also looking very awesome. Good work Chirag: https://mircle-react-git-post-functionalities-hardiegogo.vercel.app/feed

File: src/components/Sidebar.jsx (hunk 3 above)
Comment (Hardiegogo) [reply src/components/Sidebar.jsx:22]: okay

File: src/components/Post.jsx (hunk 5 above)
Comment (Hardiegogo) [reply src/components/Post.jsx:14]: Got it

**Pull request merged**